### ISSUE TITLE
FalseDiscoveryRate parameter documentation misleading / parameter can be removed

### ISSUE THREAD

**Issue by timosachsenberg** [OpenMS/OpenMS#1449]
defaults_.setValue("decoy_string", "_rev", "String which is appended at the accession of the protein to indicate that it is a decoy protein (for proteins only).");

Tool only checks for proteins if the string is contained (e.g. is a prefix). This should be reflected in the documentation

**Comment by hendrikweisser** [comment]
See also this e-mail from Florian Sigloch to the mailing list (21 May 2015):

> I wondered why there is an option to set a Decoy string in FalseDiscoveryRate. Shouldn't the PeptideIndexer already add a tag to each Decoy peptide? And if it is necessary to give the Decoy string in FalseDiscoveryRate, can the string also be used as a prefix (like in PeptideIndexer)?

In my opinion this parameter should be removed and PeptideIndexer should always be used for target/decoy annotation.

**Comment by timosachsenberg** [comment]
Discussion from mailing list
```
I wondered why there is an option to set a Decoy string in
FalseDiscoveryRate. Shouldn't the PeptideIndexer already
add a tag to each Decoy peptide? And if it is necessary to
give the Decoy string in FalseDiscoveryRate, can the string
also be used as a prefix (like in PeptideIndexer)?
```

```The string is necessary and can also be used as a prefix ("rev_" as the FalseDiscoveryRate tool only checks for its presence in the accession string of identified proteins).
For the identified peptides, the annotation from PeptideIndexer is used.
I agree that this might be a bit inconsistent between protein and peptide identifications but I think there might be some reason.
```

**Comment by timosachsenberg** [comment]
I agree

**Comment by hendrikweisser** [comment]
> Discussion from mailing list

Sorry, I saw this before I got the e-mail.

So is the parameter of FalseDiscoveryRate necessary for protein-level FDR calculations? I've added an option "annotate_proteins" to PeptideIndexer a while ago; maybe that should be used instead.

**Comment by timosachsenberg** [comment]
I only quickly looked at the code but the string seems only be used for the Protein FDR

**Comment by lars20070** [comment]
@hendrikweisser I also think both peptides and proteins should be annotated in the PeptideIndexer, and the decoy_string parameter be removed from FalseDiscoveryRate.

**Comment by timosachsenberg** [comment]
So what about making "annotate_proteins" as default and remove this flag in PeptideIndexer?
Do you see any problems that might arise if we do this?

**Comment by lars20070** [comment]
+1 IMO, the flag can be removed. So far, it only led to confusion ("Why is there no protein coverage?") since it is by default false.

**Comment by hendrikweisser** [comment]
> So what about making "annotate_proteins" as default and remove this flag in PeptideIndexer?
> Do you see any problems that might arise if we do this?

Should be fine. It's only a meta value that gets added. There could be some surprises if the target/decoy information gets lost at later steps in a pipeline, if other tools preserve only the accession strings.

> So far, it only led to confusion ("Why is there no protein coverage?") since it is by default false.

The flag doesn't have anything to do with the coverage.

**Comment by lars20070** [comment]
True @hendrikweisser. I confused it with write_protein_sequence (which I would rather see to be true by default).

**Comment by timosachsenberg** [comment]
protein sequences can be rather large so I am not sure if we should really make it default

**Comment by hendrikweisser** [comment]
While I'm working on this, I'm thinking about renaming the input parameters of FalseDiscoveryRate from "fwd_in"/"rev_in" to "in_target"/"in_decoy" - because the decoy database doesn't necessarily have to be a reversed one (could be shuffled or based on a more complicated algorithm). Along with this, a possibility would be to drop "in_target" and just have "in" (for the concatenated or target search) and "in_decoy" (for the decoy search, if necessary), i.e. get rid of one parameter.

Thoughts?

**Comment by timosachsenberg** [comment]
I think this will only break workflows that use separate target and decoy databases so I guess this is ok.
Would it make sense to make the in_decoy  an advanced parameter so it is hidden for most users? Nearly every workflow I know uses the concatenate database and it would be one additional parameter that could be hidden for the user?

**Comment by hendrikweisser** [comment]
> Would it make sense to make the in_decoy an advanced parameter so it is hidden for most users? 

Mmh, I'd prefer not to. I know some people prefer separate databases, so I'd like to keep that option obvious for them.

**Comment by timosachsenberg** [comment]
If it is really a standard case then I would also prefer separate parameters...

**Comment by hendrikweisser** [comment]
See my pull request above (#1492) for the issues discussed here. In the process I've found some other problems with FalseDiscoveryRate:
- There was an algorithm parameter ("q_value") that was "true" by default, so couldn't be used as a proper flag. I've already addressed that in the pull request.
- There is no test covering that switch (neither class nor tool test).
- There is no test covering the "algorithm:add_decoy_peptides" switch (neither class nor tool test).
- There is no tool test for the case of separate target/decoy inputs (but there are class tests).
- In the case of separate target/decoy inputs, the parameter "algorithm:add_decoy_peptides" has no effect for the TOPP tool. Even though FDRs for decoy peptides are calculated by the library class, there is no output for them.
